### PR TITLE
Remove condition "If ISBN is equal means it's a duplicate" in "duplicate check"

### LINKED ISSUE #8885: Duplicate check during import marks articles from collection as possible duplicates
### JabRef version

Latest development branch build (please note build date below)

### Operating system

GNU / Linux

### Details on version and operating system

JabRef 5.7--2022-06-01--943489e || Linux 5.10.0-14-amd64 amd64 || Java 18.0.1 || JavaFX unknown

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Copy two articles (InCollection) from the same book to a new database
2. The second entry that is copied is marked as possible duplicate although author, title, and citationkey deviate.

### Appendix

Import_erroneous_duplicates

### PR DESCRIPTION
**Pull request by AbdAlRahmanGad**
Refs #8885 

The reason of the problem is that in the 'duplicate check' if the "ISBN" is equal the entries are marked as duplicates
but that's not true in all cases. See https://github.com/JabRef/jabref/pull/9769#issuecomment-1512610441.

What I did was remove the condition and add weight to the field. The weight should be revised as I gave it a random value to demonstrate the idea. 

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [X] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/logic/database/DuplicateCheck.java [hunk 1]
```diff
@@ -65,6 +64,7 @@ public class DuplicateCheck {
         DuplicateCheck.FIELD_WEIGHTS.put(StandardField.NOTE, 0.1);
         DuplicateCheck.FIELD_WEIGHTS.put(StandardField.COMMENT, 0.1);
         DuplicateCheck.FIELD_WEIGHTS.put(StandardField.DOI, 3.);
+        DuplicateCheck.FIELD_WEIGHTS.put(StandardField.ISBN, 2.5);
```
Comment (koppor) [inline src/main/java/org/jabref/logic/database/DuplicateCheck.java:67]: Does the modification also work without this addition? Reading the comment, this adds some duplicate check for articles having the same ISBN.

I think, it's good to keep, but I am wondering... :)

Comment (AbdAlRahmanGad) [reply src/main/java/org/jabref/logic/database/DuplicateCheck.java:67]: Yes it works without this addition. Should I remove it for now?

File: src/test/java/org/jabref/logic/database/DuplicateCheckTest.java [hunk 2]
```diff
@@ -525,7 +524,9 @@ void compareOfTwoEntriesWithSameContentAndMixedLineEndingsReportsNoDifferences()
         assertTrue(duplicateChecker.isDuplicate(entryOne, entryTwo, BibDatabaseMode.BIBTEX));
     }
 
-    @Disabled("Book entries can have the same ISBN due to different chapters. The Test fails as crossref identifies both entries as the same.")
+    /**
+     *     Book entries can have the same ISBN due to different chapters. The Test fails as crossref identifies both entries as the same.
```
Comment (koppor) [inline src/test/java/org/jabref/logic/database/DuplicateCheckTest.java:528]: Use one space after `*` -- normal java doc style.

Modify the text. The test DOES NOT fail!

File: src/main/java/org/jabref/logic/database/DuplicateCheck.java (hunk 1 above)
Comment (koppor) [reply src/main/java/org/jabref/logic/database/DuplicateCheck.java:67]: Yes, remove it. We can add if we have more tests for similar entries... - I think, if entries are different, the same ISBN could make it the same. However, I am not sure about the algoirthm. Is it typos at the author? If same ISBN, the entries are same? If no ISBN, not same? That feels strange. -- Thus, remove.

**Review by koppor — APPROVED** [review]
- I added a CHANGELOG.md entry
- I changed "closes" to "refs" in the description as there is more to do (https://github.com/JabRef/jabref/issues/8885#issuecomment-2059757039)
- As is, it improves the situation. Thus, we merge
- More things, in a follow-up PR.

**Pull request merged**